main.go

```diff
@@ -349,17 +349,23 @@ func Flash(pkgName, port string, options *compileopts.Options) error {
 	return builder.Build(pkgName, fileExt, config, func(result builder.BuildResult) error {
 		// do we need port reset to put MCU into bootloader mode?
 		if config.Target.PortReset == "true" && flashMethod != "openocd" {
-			port, err := getDefaultPort(port, config.Target.SerialPort)
-			if err != nil {
-				return err
-			}
+			err := func() error {
```

Comment:
Might be cleaner to separate out this logic into a function rather than do an inline function?

Reply:
That would probably be in the form of including it in touchSerialPortAt1200bps().